core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -149,39 +159,62 @@ public void checkAccess(AlluxioURI path, CheckAccessPOptions options)
     checkUri(path);
     rpc(client -> {
       CheckAccessPOptions mergedOptions = FileSystemOptionsUtils
-          .checkAccessDefaults(mFsContext.getPathConf(path))
+          .checkAccessDefaults(getDirectAccessConf(path))
           .toBuilder().mergeFrom(options).build();
       client.checkAccess(path, mergedOptions);
       LOG.debug("Checked access {}, options: {}", path.getPath(), mergedOptions);
       return null;
     });
   }
 
+  private boolean checkDirectAccess(AlluxioURI uri) {
+    if (!getConf().isSet(PropertyKey.USER_FILE_DIRECT_ACCESS)) {
+      return false;
+    }
+    List<String> pathList = getConf().getList(PropertyKey.USER_FILE_DIRECT_ACCESS);
```

Comment:
`getConf().getList(PropertyKey.USER_FILE_DIRECT_ACCESS)` can be cached as this is called for every call and can be harmful to performance otherwise.